fs/io_zenfs.cc

```diff
@@ -229,15 +229,22 @@ ZoneFile::~ZoneFile() {
 IOStatus ZoneFile::CloseWR() {
   IOStatus s = IOStatus::OK();
 
+  s = CloseActiveZone();
+  open_for_wr_ = false;
+
+  return s;
+}
+
+IOStatus ZoneFile::CloseActiveZone() {
+  IOStatus s = IOStatus::OK();
   if (active_zone_) {
-    s = active_zone_->CloseWR();
-    ReleaseActiveZone();
-    if (s.ok()) {
-      zbd_->NotifyIOZoneClosed();
+    active_zone_->CloseWR();
```

Comment:
Doh! You're not checking the return value

Reply:
oops